CRM/Financial/BAO/Payment.php

```diff
@@ -149,6 +149,11 @@ public static function create($params) {
     elseif ($contributionStatus === 'Pending' && $params['total_amount'] > 0) {
       self::updateContributionStatus($contribution['id'], 'Partially Paid');
     }
+    elseif ($contributionStatus === 'Completed' && ((float) CRM_Core_BAO_FinancialTrxn::getTotalPayments($contribution['id'], TRUE) === 0.0)) {
```

Comment:
I might be missing something, but I'm not able to find a case where `$contributionStatus` is Completed while recording a payment via this function. It should be either `Pending` or `Pending Refund` based on amount being sent as positive or negative? @mattwire can you mention the steps on UI so I can replicate and test this change from my side? 

I tried using event registration -> change selection but I don't see any improved behavior in the contribution status.

Reply:
@jitendrapurohit This only happens when using the API function Payment.create in *refund* mode - see eg. here: https://lab.civicrm.org/extensions/stripe/blob/master/CRM/Core/Payment/StripeIPN.php#L270

Basically pass the full amount of a completed contribution in as a negative amount.

Eg. if contribution = Completed AND amount = 12 THEN
DO Payment.Create (Amount = -12)

Before this PR contribution status remains "Completed" but is fully refunded. After this PR contribution status is "Refunded" because it has been fully refunded.

@jitendrapurohit Does that help?

Reply:
Ah ok  - I've tested this and also added a UT for the same at https://github.com/civicrm/civicrm-core/pull/16255

Noting that when we refund a partial amount, the contribution status is still kept as `Completed`. This is also covered in the unit test PR.